plugins/UsersManager/API.php

```diff
@@ -264,6 +267,64 @@ private function getDefaultUserPreference($preferenceName, $login)
         }
     }
 
+    /**
+     * Returns all users with their role for $idSite.
+     *
+     * @param int $idSite
+     * @param int|null $limit
+     * @param int|null $offset
+     * @param string|null $filter_search text to search for in the user's login, email and alias (if any)
+     * @param string|null $filter_access only select users with this access to $idSite. can be 'noaccess', 'some', 'view', 'admin', 'superuser'
+     *                                   Filtering by 'superuser' is only allowed for other superusers.
+     * @return array
+     */
+    public function getUsersPlusRole($idSite, $limit = null, $offset = 0, $filter_search = null, $filter_access = null)
+    {
+        if (!$this->isUserHasAdminAccessTo($idSite)) {
+            // if the user is not an admin to $idSite, they can only see their own user
+            if ($offset > 1) {
+                return [
+                    'total' => 1,
+                    'results' => [],
+                ];
+            }
+
+            $user = $this->model->getUser($this->access->getLogin());
+            $user['role'] = $this->access->getRoleForSite($idSite);
+            $user['capabilities'] = $this->access->getCapabilitiesForSite($idSite);
+            $users = [$user];
+            $totalResults = 1;
+        } else {
```

Comment:
Can we add an `If (Piwik::hasUserSuperAccess())` here? I'm too scared that if there is any bug in the other method `$this->isUserHasAdminAccessTo($idSite)` it will return information for all users. Also is the `$this->isUserHasAdminAccessTo($idSite)` method even needed when checking here for super user access?

Reply:
Admin users have to be able to see more then their own user. The filtering based on the current user's access is in `UserTableFilter` (though could move that to the API).

Reply:
I will add a `check user has some admin access` to the top of these methods at least.